### ISSUE TITLE
react-native (w/ reactotronRedux()) freazing & crashing app 

### ISSUE THREAD

**Issue by crobinson42** [infinitered/reactotron#908]
I'm experiencing my app locking up and eventually crashing when using `reactotronRedux()` - there are no errors in my logs and this happens when an action is dispatched, but not every action. I have great test coverage in my app it has been working fine for over a year so I'm not suspicious of my codebase. 

I'd love to help debug this issue I just don't know where to start. Here's a gist for my Redux store setup and also my `ReactotronConfig.js`:

https://gist.github.com/crobinson42/2be87c7c139d9a42c1217345b9e7d5c7

```
react-native: 0.57.4
reactotron: 2.4.4 (2.4.4.154)
```

**Comment by rmevans9** [comment]
Hey @crobinson42!

This sounds like our seralization of the actions might be getting messed up on the actions that it is crashing on. Have you happened to have narrowed down which actions are causing a problem? If so having that information would go a long way in figuring this issue out.

**Comment by crobinson42** [comment]
@rmevans9 Thank you! As soon as you mentioned the serialization I went straight the issue and verified where Reactotron breaks.

We have several buttons in the app that pass a `redux-actions` `const someAction = createAction('ACTION_TYPE')` and our reducer handles the logic for those so we don't pass any second argument - low and behold, passing this to `onPress={props.someAction}` is dispatching with the event as the second argument which we discard/ignore in the reducer. That is where Reactotron is breaking.

So should Reactotron wrap the serialization in a `try/catch` for this kind of scenario?

**Comment by rmevans9** [comment]
Hmm... the `onPress` event gives you a SyntheticEvent. I wonder if there is something in there blowing up. Booting up the example app to try and replicate this issue. I feel like we should make sure `reactotron` works with this correctly because I feel like this is a common enough use case.

A quick workaround to test this would be to change your onPress to be something like `onPress={() => { props.someAction(); }}`. This would stop passing the event through the action creator and tell us if this is truly the problem.

**Comment by crobinson42** [comment]
Yep, that’s exactly how I verified it was the SyntheticEvent blowing it up using the same as your example in our code (` onPress={() => { props.someAction(); }}`)

**Comment by rmevans9** [comment]
Perfect. I am going to replicate and see what can be done to get this working better without having to apply that workaround.

**Comment by rmevans9** [comment]
So as a kinda hack work around I am going to be pushing out a fix to `reactotron-core-client` and `reactotron-react-native` that should solve this. Transferring this to `reactotron-react-native` since that is where the real fix will take place.
